test/Storage.js

```diff
@@ -0,0 +1,160 @@
+        expect(response).to.be.an('array');
+      });
+
+      it('should returns an array of data objects', async () => {
+        const response = await storage.listData();
+        expect(response).to.deep.equal(dataList);
+      });
+    });
+
+    describe('with date interval query', () => {
+      let dataBetweenTimeInterval;
+
+      beforeEach(() => {
+        query = { startDate: '2019-03-18T14:42:03.192Z', finishDate: '2019-03-18T16:17:45.192Z'};
+        dataBetweenTimeInterval = dataList.slice(1, 4);
+        storage = new Storage(settings);
+        nock(url)
+          .get(`/data`)
+          .query(query)
+          .reply(200, dataBetweenTimeInterval);
+        });
+
+      it('should returns an array', async () => {
+        const response = await storage.listData(query);
+        expect(response).to.be.an('array');
+      });
+
+      it('should returns an array of data objects between time interval', async () => {
+        const response = await storage.listData(query);
+        expect(response).to.deep.equal(dataBetweenTimeInterval);
+      });
+    });
+
+    describe('with take and skip query', () => {
+      let skippedData;
+
+      beforeEach(() => {
+        query = { skip: 3, take: 2 };
+        skippedData = dataList.slice(query.skip, dataList.length);
+        storage = new Storage(settings);
+        nock(url)
+          .get('/data')
+          .query(query)
+          .reply(200, skippedData);
+      });
+
+      it('should returns an array', async () => {
+        const response = await storage.listData(query);
+        expect(response).to.be.an('array');
+      });
+
+      it('should returns an array of two data objects skipped by 3', async () => {
+        const response = await storage.listData(query);
+        expect(response).to.deep.equal(skippedData);
+      });
+    });
+
+    describe('with order and orderBy query', () => {
+      let reversedData;
+
+      beforeEach(() => {
+        query = { order: -1, orderBy: 'timestamp' };
+        reversedData = dataList.reverse();
+        storage = new Storage(settings);
+        nock(url)
+          .get('/data')
+          .query(query)
+          .reply(200, reversedData);
+      });
+
+      it('should returns an array', async () => {
+        const response = await storage.listData(query);
+        expect(response).to.be.an('array');
+      });
+
+      it('should returns an array of data objects descending ordered by timestamp', async () => {
+        const response = await storage.listData(query);
+        expect(response).to.deep.equal(reversedData);
+      });
+    });
+
+    describe('when an error happens', () => {
+      it('should pass error to client', async () => {
+        storage = new Storage(settings);
+        nock(url)
+          .get('/data')
+          .reply(500);
+
+          try {
+            await storage.listData();
+          } catch (err) {
+            expect(err).to.have.property('isAxiosError', true);
```

Comment:
`isAxiosError` ir fine, but could you verify that it is simply an `Error`? If you can, it doesn't force the library to use axios.

Reply:
Done.